### PR TITLE
Use Chef::DataBag instead of ProposalObject for barclamp info

### PR DESCRIPTION
**Pull request by bkutil**
Refs https://github.com/crowbar/barclamp-crowbar/pull/1269

### CODE REVIEW

File: crowbar_framework/app/models/provisioner_service.rb [hunk 1]
```diff
@@ -143,7 +143,7 @@ def transition(inst, name, state)
         else
           node_barclamp = node_role
         end
-        bc_databag = ProposalObject.find_data_bag_item("barclamps/#{node_barclamp}")
+        bc_databag = Chef::DataBag.load("barclamps/#{node_barclamp}") rescue nil
```
Comment (vuntz) [inline crowbar_framework/app/models/provisioner_service.rb]: That's not correct (and broken with the move away from data bags).

When we discussed it the other day, though, we found out that we can just drop this whole block (`if state == "installing"`) as no barclamp is using a `unsupported_platform` attribute.

Comment (bkutil) [inline crowbar_framework/app/models/provisioner_service.rb]: > That's not correct (and broken with the move away from data bags).

Broken as it should rescue to an array? Or in what sense?

Comment (vuntz) [inline crowbar_framework/app/models/provisioner_service.rb]: Broken as the data bag should not exist anymore anyway :-)

Comment (tboerger) [inline crowbar_framework/app/models/provisioner_service.rb]: Did i remember correctly that we don't provide the unsupported platform attribute anymore?

Comment (vuntz) [inline crowbar_framework/app/models/provisioner_service.rb]: @tboerger yes, hence my suggestion to simply drop the whole block :-)

Comment (bkutil) [inline crowbar_framework/app/models/provisioner_service.rb]: @vuntz just to be clear: so everything inside 'installing' if block can go away?

Comment (tboerger) [inline crowbar_framework/app/models/provisioner_service.rb]: > @tboerger yes, hence my suggestion to simply drop the whole block :-)

Makes sense
:+1:

Comment (vuntz) [inline crowbar_framework/app/models/provisioner_service.rb]: @bkutil yes

Comment (bkutil) [inline crowbar_framework/app/models/provisioner_service.rb]: > @bkutil yes

Dropped.

File: crowbar_framework/app/models/provisioner_service.rb [hunk 2]
```diff
@@ -76,7 +76,7 @@ def transition(inst, name, state)
         add_role_to_instance_and_node("provisioner", inst, name, db, role, "provisioner-server")
 
         # Reload the roles
-        db = ProposalObject.find_proposal "provisioner", inst
+        db = db.reload
```
Comment (vuntz) [inline crowbar_framework/app/models/provisioner_service.rb]: Just `db.reload`, maybe?

Comment (bkutil) [inline crowbar_framework/app/models/provisioner_service.rb]: > Just `db.reload`, maybe?

Yep, fixed.

**Pull request merged**